schema/queries.gql

```diff
@@ -155,6 +155,42 @@ type Query {
     """
     dangling: Boolean = false
   ): CompositionConnection!
+
+  """
+  Get an `KubernetesResource` and its descendants which form a tree. The two
+  `KubernetesResource`s that have descendants are `CompositeResourceClaim` (its
+  `CompositeResource`) and `CompositeResource` (the `KubernetesResource`s it
+  composes via a `Composition`).
+  """
+  crossplaneResourceTree(
+    "The `ID` of an `CrossplaneResource`"
+    id: ID!
+  ): CrossplaneResourceTreeConnection!
+}
```

Comment:
I imagine this could also be a field on claims, XRs, etc like `children` or `exploded` rather than a distinct query. Would that offer a better or worse UX?

Reply:
That would work but it could result in asking for a tree for every object in an array or even worse asking for a tree for every element in a tree. By putting it at a top level we encourage it to be used more conservatively.